### PR TITLE
Move AppCenter Analytics validation to shouldFilter callback

### PR DESCRIPTION
**Pull request by MatkovIvan**

### CODE REVIEW

File: AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m [hunk 1]
```diff
@@ -187,6 +187,25 @@ + (BOOL)isAutoPageTrackingEnabled {
 
 #pragma mark - Private methods
 
+- (BOOL)validateLog:(MSLogWithNameAndProperties *)log {
```
Comment (clpolet) [inline AppCenterAnalytics/AppCenterAnalytics/MSAnalytics.m]: I would say we can get rid of this method and do everything in the shouldFilter.

File: AppCenterAnalytics/AppCenterAnalytics/Internals/Model/MSEventLog.m [hunk 2]
```diff
@@ -4,7 +4,6 @@
 static NSString *const kMSTypeEvent = @"event";
 
 static NSString *const kMSId = @"id";
-static NSString *const kMSName = @"name";
```
[the remark this answers is not in the document]
Comment (clpolet) [reply AppCenterAnalytics/AppCenterAnalytics/Internals/Model/MSEventLog.m]: We'll update the schema later.

File: AppCenterAnalytics/AppCenterAnalytics/Internals/MSAnalytics+Validation.m [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+#import "MSPageLog.h"
+
+// Events values limitations
+static const int minEventNameLength = 1;
+static const int maxEventNameLength = 256;
+
+@implementation MSAnalytics (Validation)
+
+- (BOOL)shouldFilterLog:(id<MSLog>)log {
+  NSObject *logObject = (NSObject *)log;
```
Comment (clpolet) [inline AppCenterAnalytics/AppCenterAnalytics/Internals/MSAnalytics+Validation.m]: I tried to put a break point here but it never got hit. Is the delegate registered?

Comment (MatkovIvan) [reply AppCenterAnalytics/AppCenterAnalytics/Internals/MSAnalytics+Validation.m]: Yeah, this is linker issue in the test app. I've added a workaround such as we use in other categories.

**Pull request merged**